R/process_source.R

```diff
@@ -1,131 +1,71 @@
-  # - "OBJECT_DEFINITION": We're inside an object definition (usually a
-  #     function).
-  # - "LEADING_COMMENT": We're in a comment that (probably) precedes an object
-  #     definition.
-  chunks <- list()
-  state <- "SCANNING"
-  current_chunk <- vec_accumulator()
-  current_chunk_name <- NULL
-
-  for (line in text) {
-    if (state == "SCANNING") {
-      if (grepl("^\\S+ *<-", line)) {
-        # Lines that start with "foo <-" tell us that the object definition
-        # starts here.
-        state <- "OBJECT_DEFINITION"
-        current_chunk$add(line)
-        current_chunk_name <- extract_object_name(line)
-
-        # Special case: check if it's a one-liner (one-liners will be parseable;
-        # an opening line for multi-line functions/expressions will not.). If
-        # so, transition to SCANNING state.
-        if (is_parseable(line)) {
-          state <- "SCANNING"
-          chunks[[current_chunk_name]] <- current_chunk$get()
-          current_chunk$reset()
-          current_chunk_name <- NULL
-        }
-
-      } else if (grepl("^#", line)) {
-        state <- "LEADING_COMMENT"
-        current_chunk$add(line)
-      }
-
-    } else if (state == "OBJECT_DEFINITION") {
-      current_chunk$add(line)
-      if (grepl("^\\S", line)) {
-        # If the line starts with anything _other_ than whitespace, then we know
-        # we've reached the end of the object definition.
-        state <- "SCANNING"
-        chunks[[current_chunk_name]] <- current_chunk$get()
-        current_chunk$reset()
-        current_chunk_name <- NULL
-      }
-
-    } else if (state == "LEADING_COMMENT") {
-      if (grepl("^\\S+ *<-", line)) {
-        # Lines that start with "foo <-" tell us that the object definition
-        # starts here.
-        state <- "OBJECT_DEFINITION"
-        current_chunk$add(line)
-        current_chunk_name <- extract_object_name(line)
-
-        # Special case: check if it's a one-liner (one-liners will be parseable;
-        # an opening line for multi-line functions/expressions will not.). If
-        # so, transition to SCANNING state.
-        if (is_parseable(line)) {
-          state <- "SCANNING"
-          chunks[[current_chunk_name]] <- current_chunk$get()
-          current_chunk$reset()
-          current_chunk_name <- NULL
-        }
-
-      } else if (grepl("^#", line)) {
-        # Stay in current state.
-        current_chunk$add(line)
-
-      } else {
-        # We've hit the end of a comment block but no definition here. Just
-        # discard the block and go back to scanning mode.
-        state <- "SCANNING"
-        current_chunk$reset()
-      }
-    }
+# Given a file, process the text and return a named list of objects.
+process_source_file_one <- function(file) {
+  parse_data <- utils::getParseData(parse(file = file, keep.source = TRUE))
+  text <- readLines(file)
+
+  assignment_ops <- parse_data[
+    parse_data$token %in% c("LEFT_ASSIGN", "EQ_ASSIGN", "RIGHT_ASSIGN"),
```

Comment:
Can you remove `RIGHT_ASSIGN`? It obviously is valid R code, but I don't think it's necessary to support it for staticimports, and I don't want to encourage people to use it.

Reply:
Got it, [ed9f10e](https://github.com/wch/staticimports/pull/21/commits/ed9f10e3f09f9fd44fdbd74ecc93542245b571fb) removes support for `->`